lizard_auth_server/views_api_v2.py

```diff
@@ -0,0 +1,227 @@
+        data['permissions'].append({
+            'content_type': perm.content_type.natural_key(),
+            'codename': perm.codename,
+        })
+
+    # TODO: not sure if needed, but why not..
+    data['company'] = str(profile.company)
+
+    # datetimes should be serialized to an iso8601 string
+    data['created_at'] = profile.created_at.isoformat()
+    return data
+
+
+class AuthorizeView(FormInvalidMixin, ProcessGetFormView):
+    form_class = forms.JWTDecryptForm
+
+    def form_valid(self, form):
+        self.domain = get_domain(form)
+        self.site = form.site
+        if self.request.user.is_authenticated():
+            return self.form_valid_authenticated()
+        return self.form_valid_unauthenticated(
+            form.cleaned_data.get('message', {}).get('force_sso_login', True))
+
+    def form_valid_authenticated(self):
+        """
+        Called when login succeeded.
+        """
+        if self.has_access():
+            return self.success()
+        return self.access_denied()
+
+    def build_login_url(self):
+        """
+        Store the authorize view (most likely the current view) as
+        "next" page for a login page.
+        """
+        nextparams = {
+            'message': self.request.GET['message'],
+            'key': self.request.GET['key'],
+        }
+        params = urlencode([(
+            'next',
+            '%s?%s' % (
+                reverse('lizard_auth_server.api_v2.authorize'),
+                urlencode(nextparams))
+        )])
+        return '%s?%s' % (reverse('django.contrib.auth.views.login'), params)
+
+    def build_back_to_portal_url(self):
+        """Redirect user back to the portal, without logging him in."""
+        return urljoin(self.domain, 'sso/local_not_logged_in/')
+
+    def form_valid_unauthenticated(self, force_sso_login):
+        """
+        Redirect user to login page if force_sso_login == True, else, return
+        without having to log in.
+        """
+        if force_sso_login:
+            # Typical situation -- force the user to login.
+            return HttpResponseRedirect(self.build_login_url())
+        else:
+            # Return the unauthenticated user back to the portal.
+            return HttpResponseRedirect(self.build_back_to_portal_url())
+
+    def has_access(self):
+        """
+        Check whether the user has access to the site.
+        """
+        if not self.request.user.is_active:
+            # extra check: should not be necessary as inactive users can't
+            # login anyway
+            return False
+        try:
+            profile = self.request.user.profile
+        except Profile.DoesNotExist:
+            return False
+        return profile.has_access(self.site)
+
+    def success(self):
+        payload = {
+            'key': self.site.sso_key,
+            # Dump all relevant data:
+            'user': json.dumps(construct_user_data(self.request.user)),
```

Comment:
Ah, hier wordt de construct_user_data gebruikt.

In geen enkele site wordt als het goed is nog gekeken naar de is_superuser en is_staff status die iemand in de SSO eventueel heeft. Dus dat hoeft niet doorgegeven te worden. Idem de rechten. Anders wordt het denk ik ook te groot.